### PR TITLE
[tests] Use a different filename for the invalid hostname workflows test

### PR DESCRIPTION
**Pull request by townsend2010**

### CODE REVIEW

File: tests/test_data/workflows/v1/42-invalid-hostname-workflow.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-description: Invalid file name workflow
+description: Invalid host name workflow
 version: 0.1
 
 instances:
-  invalid-$filename-workflow:
+  42- invalid-hostname-workflow:
```
Comment (Saviq) [inline tests/test_data/workflows/v1/42-invalid-hostname-workflow.yaml:5]: Space on purpose?

File: tests/test_workflow_provider.cpp [hunk 2]
```diff
@@ -286,13 +286,13 @@ TEST_F(VMWorkflowProvider, allWorkflowsReturnsExpectedInfo)
...
     logger_scope.mock_logger->expect_log(
-        mpl::Level::error, "Invalid workflow name \'invalid-\?filename-workflow\': must be a valid host name");
+        mpl::Level::error, "Invalid workflow name \'42-invalid-hostname-workflow\': must be a valid host name");
 
     mp::DefaultVMWorkflowProvider workflow_provider{workflows_zip_url, &url_downloader, cache_dir.path(), default_ttl};
 
     auto workflows = workflow_provider.all_workflows();
 
-    EXPECT_EQ(workflows.size(), 11ul);
+    EXPECT_EQ(workflows.size(), 10ul);
```
Comment (Saviq) [inline tests/test_workflow_provider.cpp:295]: Hmm why this change?

File: tests/test_data/workflows/v1/42-invalid-hostname-workflow.yaml (hunk 1 above)
Comment (townsend2010) [reply tests/test_data/workflows/v1/42-invalid-hostname-workflow.yaml]: Nope :smiley: Fixed

File: tests/test_workflow_provider.cpp (hunk 2 above)
Comment (townsend2010) [reply tests/test_workflow_provider.cpp:295]: Yeah, so I recreated the zip file and it seems there was a left over file in the old zip that was no longer valid in the `test_data/workflows/v1/` directory, so the number of workflows found decreased by one.

**Review by Saviq — APPROVED** [review]
Yup.

bors merge
